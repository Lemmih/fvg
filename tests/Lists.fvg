data List a = Nil | Cons a (List a)

map : (a -> b) -> List a -> List b
map fn lst =
  case lst of
    Nil -> Nil
    Cons x xs -> Cons (fn x) (map fn xs)

main : Svg
main =
  let x = map (\x -> plus x 1) (Cons 10 (Cons 20 Nil)) in
  <svg>
    <text>"Hello World: {{x}}"</text>
  </svg>
