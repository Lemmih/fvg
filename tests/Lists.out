 <svg > <text >"Hello World: Cons 11 (Cons 21 Nil)" </text>  </svg> 